src/provider/cloud-element-templates/ElementTemplates.js

```diff
@@ -50,8 +57,13 @@ export default class ElementTemplates extends DefaultElementTemplates {
    */
   applyTemplate(element, newTemplate) {
 
+    let action = 'apply';
     const oldTemplate = this.get(element);
 
+    if (newTemplate && oldTemplate && (newTemplate.id === oldTemplate.id)) {
+      action = 'update';
```

Comment:
We'd want to fire `remove`, if we un-apply the template (no new template is given)?